apis/server/image_bridge.go

```diff
@@ -76,13 +77,32 @@ func (s *Server) searchImages(ctx context.Context, rw http.ResponseWriter, req *
 	return EncodeResponse(rw, http.StatusOK, searchResultItem)
 }
 
-// removeImage deletes an image by reference
+// removeImage deletes an image by reference.
 func (s *Server) removeImage(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
 	name := mux.Vars(req)["name"]
 
-	option := &mgr.ImageRemoveOption{}
+	image, err := s.ImageMgr.GetImage(ctx, name)
+	if err != nil {
+		return err
+	}
+
+	containers, err := s.ContainerMgr.List(ctx, func(c *types.ContainerInfo) bool {
```

Comment:
Actually I think this part should not exist in bridge layer. Instead, it should exist in mgr layer. Because it has already coupled with inner logic of store and mgr operations.

Reply:
First time, i also think put this part in mgr, but in ImageMgr, if i want to check whether a image is being used by a container, i must add a reference of ContainerMgr, it may make the logic a little bit complexed. WDYT @allencloud

Reply:
It is a tradeoff.  I think we in the future there will be more interactions among all managers in pouch daemon. Like container manager and network manager, container manager and volume manager, and so on.

The interaction is no doubt, how to design efficiently among managers' interaction is the next step we need to do.

Reply:
Yes, i agree with you, we may need redesign the daemon Init

Reply:
Could you please solve the conflict first?
Let this PR be merged first and maybe a follow-up to handle what we communicated just now.

Reply:
Ok